garrysmod/lua/includes/extensions/string.lua

```diff
@@ -345,3 +345,44 @@ function string.Comma( number )
 	return number
 
 end
+
+--[[---------------------------------------------------------
+	Name: NiceNumber( num )
+	Desc: Abbreviates numbers and adds a suffix	
+	Examples: string.NiceNumber( 69420 ) ==> "69.42K"
+			string.NiceNumber( -123456789 ) ==> "-123.46M"
+-----------------------------------------------------------]]
+local nice_number_suffixes = {
+    "K","M","B","T","Qa",
+    "Qi","Sx","Sp","Oc","No",
+    "Dc","Ud","Dd","Td","Qad",
+    "Qid","Sxd","Spd","Ocd","Nod",
+    "Vg","Uvg","Dvg","Tvg","Qavg",
+    "Qivg","Sxvg","Spvg","Ocvg"
+}
+
+function string.NiceNumber( num )
+    local suffixid, absNum, isNegative = nil, tonumber( num ), false
+    if not absNum then return num end
+    
+    isNegative = absNum < 0
+    absNum = math.abs( absNum )
+
+    if absNum == math.huge then
+        return string.format( "%sInfinity", isNegative and "-" or "" )
+    end
+
+    for i = 1, #nice_number_suffixes do
+        if absNum >= ( 10 ^ ( i * 3 ) ) then
+            suffixid = i
+        else
+            break
+        end
+    end

```

Comment:
Perhaps this is more efficient, not nested:
```suggestion
    if ( absNum >= 1000 ) then
        suffixid = math.min( #nice_number_suffixes, math.floor( math.log10( absNum ) / 3 ) )
    end
```
Also, you should indent using tabs to be consistent with the code base.